src/lightning/pytorch/strategies/xla_fsdp.py

```diff
@@ -386,6 +388,9 @@ def save_checkpoint(
         # this is the output of `self.lightning_module_state_dict`
         state_dict = checkpoint.pop("state_dict")
 
+        # not needed in ckpt consolidation
+        state_dict["optimizer"] = checkpoint.pop("optimizer_states")
```

Comment:
Why is this done? Shouldn't the optimizer states be included in the "metadata" checkpoint? Because the logic for loading them is not customized at all

Reply:
This is done to match the torch_xla mnist tutorial: https://github.com/pytorch/xla/blob/master/test/test_train_mp_mnist_fsdp_with_ckpt.py#L288-L290, where the checkpoint being saved is defined as 
```
ckpt = {
        'model': model.state_dict(),
        'shard_metadata': model.get_shard_metadata(),
        'optimizer': optimizer.state_dict(),  # not needed in ckpt consolidation
    }
 ```
 from https://github.com/Lightning-AI/lightning/blob/master/src/lightning/pytorch/strategies/strategy.py#L165-L181 I think `checkpoint.pop("optimizer_states")` will return the optimizer.state_dict() right?